File: routes/blogs.js
const express = require('express');
const { getBlogs, getBlog, createBlog, commentOnBlog } = require('../controllers/blogController');
const auth = require('../middlewares/auth');

const router = express.Router();

router.get('/', getBlogs);
router.get('/:id', getBlog);
router.post('/', auth, createBlog);
router.post('/:id/comments', auth, commentOnBlog);

module.exports = router;

File: routes/blogs.go
package routes

import (
	"github.com/gin-gonic/gin"
	"Backend/controllers"
	"Backend/middlewares"
)

func BlogRoutes(router *gin.RouterGroup) {
	router.GET("/", controllers.GetBlogs)
	router.GET("/:id", controllers.GetBlog)
	router.POST("/", middlewares.Auth(), controllers.CreateBlog)
	router.POST("/:id/comments", middlewares.Auth(), controllers.CommentOnBlog)
}
